### PR TITLE
Feature/apache2 4 config

### PR DESCRIPTION
**Pull request by whitx**
# Description

Added conditions for apache version in httpd config files
# Impacts

Httpd services
# Delete branch after merge

NO
# NEWS file entries
## Enhancements
- Redo the conditioning for apache 2-4

### CODE REVIEW

File: addons/packages/packetfence.spec [hunk 1]
```diff
@@ -948,6 +948,7 @@ fi
                         /usr/local/pf/conf/haproxy.conf.example
 %dir                    /usr/local/pf/conf/httpd.conf.d
 %config                 /usr/local/pf/conf/httpd.conf.d/captive-portal-common.conf
+%config                 /usr/local/pf/conf/httpd.conf.d/captive-portal-common2-4.conf
```
Comment (extrafu) [inline addons/packages/packetfence.spec]: I think I would prefer it to be called captive-portal-common24.conf

File: conf/httpd.conf.d/httpd.portal.tt.example [hunk 2]
```diff
@@ -114,6 +148,9 @@ PerlSwitches -I[% install_dir %]/html/captive-portal/lib
 # portal itself.
 PerlPostConfigRequire [% install_dir %]/lib/pf/web/captiveportal_modperl_require.pl
 PerlLoadModule captiveportal
+[% IF apache_version == "2.4" %]
+PerlLoadModule pf::web::apache2_version
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.portal.tt.example:154]: Add this module to this PR

Comment (whitx) [inline conf/httpd.conf.d/httpd.portal.tt.example:154]: Done

File: lib/pf/services/manager/httpd.pm [hunk 3]
```diff
@@ -181,10 +181,11 @@ sub generateCommonConfig {
     parse_template( \%tags, "$conf_dir/httpd.conf.d/ssl-certificates.conf", "$generated_conf_dir/ssl-certificates.conf", "#" );
 
     # TODO we *could* do something smarter and process all of conf/httpd.conf.d/
-    my @config_files = ( 'captive-portal-common.conf');
+    my @config_files = ( 'captive-portal-common.tt.example');
```
Comment (fdurand) [inline lib/pf/services/manager/httpd.pm]: not tt.example

File: addons/packages/packetfence.spec [hunk 4]
```diff
@@ -947,7 +947,8 @@ fi
 %config(noreplace)      /usr/local/pf/conf/haproxy.conf
                         /usr/local/pf/conf/haproxy.conf.example
 %dir                    /usr/local/pf/conf/httpd.conf.d
-%config                 /usr/local/pf/conf/httpd.conf.d/captive-portal-common.conf
+%config                 /usr/local/pf/conf/httpd.conf.d/captive-portal-common.tt
+%config                 /usr/local/pf/conf/httpd.conf.d/captive-portal-common.tt.example
```
Comment (fdurand) [inline addons/packages/packetfence.spec]: not a %config

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 5]
```diff
@@ -1,1003 +1,149 @@
...
-# require it to be present in the URL.  So "/icons" isn't aliased in this
-# example, only "/icons/".  If the fakename is slash-terminated, then the 
-# realname must also be slash terminated, and if the fakename omits the 
-# trailing slash, the realname must also omit it.
-#
-# We include the /icons/ alias for FancyIndexed directory listings.  If you
-# do not use FancyIndexing, you may comment this out.
-#
-Alias /icons/ "/var/www/icons/"
+ErrorLog [% install_dir %]/logs/httpd.parking.error
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: [ %name% ]

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 6]
```diff
@@ -1,1003 +1,149 @@
...
+LogLevel warn
 
-#
-# WebDAV module configuration section.
-# 
-<IfModule mod_dav_fs.c>
-    # Location of the WebDAV lock database.
-    DAVLockDB /var/lib/dav/lockdb
-</IfModule>
+CustomLog [% install_dir %]/logs/httpd.parking.access combined
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: [ %name% ]

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 7]
```diff
@@ -1,1003 +1,149 @@
...
-# server will use that explicit path.  If the filenames do *not* begin
-# with "/", the value of ServerRoot is prepended -- so "logs/foo.log"
-# with ServerRoot set to "/etc/httpd" will be interpreted by the
-# server as "/etc/httpd/logs/foo.log".
-#
+#PacketFence apache configuration file
+
+#Debian specific
+<IfDefine debian>
+  <IfModule !mod_perl.c>
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: remove

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 8]
```diff
@@ -1,1003 +1,149 @@
...
+      LoadModule socache_shmcb_module /usr/lib/apache2/modules/mod_socache_shmcb.so
+    </IfModule>
+    <IfModule !prefork.c>
+      LoadModule mpm_prefork_module /usr/lib/apache2/modules/mod_mpm_prefork.so
+    </IfModule>
+  [% END %]
+  <IfModule !mod_log_config.c>
+    LoadModule log_config_module /usr/lib/apache2/modules/mod_log_config.so
+  </IfModule>
+  <IfModule !mod_ssl.c>
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: remove

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 9]
```diff
@@ -1,1003 +1,149 @@
...
+    LoadModule ssl_module /usr/lib/apache2/modules/mod_ssl.so
+  </IfModule>
+  <IfModule !mod_alias.c>
+    LoadModule alias_module /usr/lib/apache2/modules/mod_alias.so
+  </IfModule>
+  <IfModule !mod_mime.c>
+    LoadModule mime_module /usr/lib/apache2/modules/mod_mime.so
+  </IfModule>
+  <IfModule !mod_proxy.c>
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: remove

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 10]
```diff
@@ -1,1003 +1,149 @@
...
+    LoadModule proxy_module /usr/lib/apache2/modules/mod_proxy.so
+  </IfModule>
+  <IfModule !proxy_http.c>
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: remove

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 11]
```diff
@@ -1,1003 +1,149 @@
...
+    LoadModule proxy_http_module /usr/lib/apache2/modules/mod_proxy_http.so
+  </IfModule>
+  <IfModule !mod_apreq2.c>
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: remove

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 12]
```diff
@@ -1,1003 +1,149 @@
...
+    LoadModule apreq_module /usr/lib/apache2/modules/mod_apreq2.so
+  </IfModule>
+  <IfModule !mod_headers.c>
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: not sure if we use it

File: conf/httpd.conf.d/httpd.portal.tt.example [hunk 13]
```diff
@@ -240,27 +297,43 @@ NameVirtualHost [% vhost %]:443
      DocumentRoot [% install_dir %]/html/captive-portal/lib
      ErrorLog [% install_dir %]/logs/[% name %].error
      CustomLog [% install_dir %]/logs/[% name %].access combined env=!is_collectd
-     Include [% var_dir %]/conf/captive-portal-common.conf
+     Include [% var_dir %]/conf/captive-portal-common.tt
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.portal.tt.example]: Include [% var_dir %]/conf/captive-portal-common

File: conf/httpd.conf.d/httpd.portal.tt.example [hunk 14]
```diff
@@ -288,27 +369,43 @@ NameVirtualHost [% vhost %]:443
      DocumentRoot [% install_dir%]/html/captive-portal/lib
      ErrorLog [% install_dir %]/logs/[% name %].error
      CustomLog [% install_dir %]/logs/[% name %].access combined env=!is_collectd
-     Include [% var_dir %]/conf/captive-portal-common.conf
+     Include [% var_dir %]/conf/captive-portal-common.tt
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.portal.tt.example]: Include [% var_dir %]/conf/captive-portal-common

File: lib/pf/services/manager/httpd.pm [hunk 15]
```diff
@@ -181,11 +182,10 @@ sub generateCommonConfig {
...
     # TODO we *could* do something smarter and process all of conf/httpd.conf.d/
-    my @config_files = ( 'captive-portal-common.conf');
-    foreach my $config_file (@config_files) {
-        $logger->info("generating $generated_conf_dir/$config_file");
-        parse_template(\%tags, "$conf_dir/httpd.conf.d/$config_file", "$generated_conf_dir/$config_file");
-    }
+    my $config_file = "captive-portal-common";
+    my $tt = Template->new(ABSOLUTE => 1);
+    $logger->info("generating $generated_conf_dir/$config_file");
+    $tt->process("$conf_dir/httpd.conf.d/$config_file.tt", \%tags, "$generated_conf_dir/$config_file") or die $tt->error();
```
Comment (fdurand) [inline lib/pf/services/manager/httpd.pm]: $vars

File: conf/httpd.conf.d/httpd.parking.tt.example [hunk 16]
```diff
@@ -1,1003 +1,125 @@
...
-# CGI directory exists, if you have that configured.
-#
-<Directory "/usr/local/pf/html/parking/cgi-bin">
-    AllowOverride None
-    Options None
-    Order allow,deny
-    Allow from all
+<Directory [% install_dir%]/html/parking/cgi-bin>
+  [% IF apache_version == "2.4" %]
+  Require granted all
```
Comment (fdurand) [inline conf/httpd.conf.d/httpd.parking.tt.example]: Require all granted

**Pull request merged**